code/modules/vote/vote_datum.dm

```diff
@@ -59,21 +60,32 @@
 
 // Returns the result
 /datum/vote/proc/calculate_result()
-	switch(vote_result_type)
-		if(VOTE_RESULT_TYPE_MAJORITY)
+	// Sanitize the votes first
+	for(var/ck in voted)
+		if(!(voted[ck] in choices))
+			stack_trace("Someone managed to cast a vote outside of presented options! The choices were [json_encode(choices)]; the vote is [voted[ck]]")
+			voted -= ck  // so that the following code doesn't get confused
+			continue
+
+	// Count up all votes
+	var/list/results = list()
+	for(var/choice in choices)
+		results[choice] = 0
+	for(var/ck in voted)
+		results[voted[ck]] += 1
+
+	// And log them to DB
+	if(!is_custom)
+		for(var/res in results)
+			SSblackbox.record_feedback("nested tally", "votes", results[res], list(vote_type_text, res), ignore_seal = TRUE)
+
+	// Here goes the actual winner selection code
+	switch(vote_counting)
+		if(VOTE_COUNTING_MAJORITY)
 			if(!length(voted))
```

Comment:
If we check this twice should we just move it to the common area?

Reply:
No because it's feasible to have a `VOTE_COUNTING_1984` in which votes don't actually matter, and lack of votes would still result in the win of The Correct Candidate.